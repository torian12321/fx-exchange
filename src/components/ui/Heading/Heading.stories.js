import React from 'react';
import { Heading } from './Heading';

export default {
  title: 'Components|Atoms/Heading'
};

export const basic = () => <Heading caption='Header Title' />;

export const customLevel = () => (
  <>
    <Heading caption='This is header 1' level={1} />
    <Heading caption='This is header 2' level={2} />
    <Heading caption='This is header 3' level={3} />
    <Heading caption='This is header 4' level={4} />
    <Heading caption='This is header 5' level={5} />
    <Heading caption='This is header 6' level={6} />
  </>
);
