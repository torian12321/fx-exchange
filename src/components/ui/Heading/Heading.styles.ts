import { css } from '@emotion/core'
import { theme } from 'assets/styles';

export const heading = css`
  margin: .5em 0;
  display: block;
  font-weight: bold;
  color: ${theme.colors.text.dark};
  
  h1&,
  h2& {
    color: ${theme.colors.primary};
  }
  
  h1& { font-size: 2.4em; }
  h2& { font-size: 2em; }
  h3& { font-size: 1.8em; }
  h4& { font-size: 1.25em; }
  h5& { font-size: 1.2em; }
  h6& { font-size: 1em; }
`;