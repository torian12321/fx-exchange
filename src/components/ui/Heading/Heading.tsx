import React from "react";
import classNames from "classnames";
/** @jsx jsx */
import { jsx } from '@emotion/core'
import { IHeading } from "./Heading.interfaces";
import * as styles from "./Heading.styles";

const Heading = (props: IHeading) => {
  const { children, caption, level = 1, className } = props;
  const content = children || caption;
  const TagName: any = `h${level}`;

  return (
    <TagName
      css={styles.heading}
      className={classNames(className)}
    >
      {content}
    </TagName>
  );
};

export { Heading };
