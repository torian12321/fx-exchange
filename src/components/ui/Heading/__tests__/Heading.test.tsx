import * as React from "react";
import serializer from 'jest-emotion'
import { create } from "react-test-renderer";
import { Heading } from "../Heading";

expect.addSnapshotSerializer(serializer)

describe("UI Component: Header", () => {
  [1, 2, 3, 4, 5, 6].forEach(level =>
    it(`should correctly render h${level} for level ${level}`, () => {
      const heading = create(
        <Heading caption="Header text" level={level} />
      );

      expect(heading.toJSON()).toMatchSnapshot();
    })
  );
});
