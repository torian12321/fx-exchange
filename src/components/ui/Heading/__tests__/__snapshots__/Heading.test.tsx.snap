// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Component: Header should correctly render h1 for level 1 1`] = `
.emotion-0 {
  margin: .5em 0;
  display: block;
  font-weight: bold;
  color: #0074ea;
}

h1.emotion-0,
h2.emotion-0 {
  color: #0074ea;
}

h1.emotion-0 {
  font-size: 2.4em;
}

h2.emotion-0 {
  font-size: 2em;
}

h3.emotion-0 {
  font-size: 1.8em;
}

h4.emotion-0 {
  font-size: 1.25em;
}

h5.emotion-0 {
  font-size: 1.2em;
}

h6.emotion-0 {
  font-size: 1em;
}

<h1
  className=" emotion-0"
>
  Header text
</h1>
`;

exports[`UI Component: Header should correctly render h2 for level 2 1`] = `
.emotion-0 {
  margin: .5em 0;
  display: block;
  font-weight: bold;
  color: #0074ea;
}

h1.emotion-0,
h2.emotion-0 {
  color: #0074ea;
}

h1.emotion-0 {
  font-size: 2.4em;
}

h2.emotion-0 {
  font-size: 2em;
}

h3.emotion-0 {
  font-size: 1.8em;
}

h4.emotion-0 {
  font-size: 1.25em;
}

h5.emotion-0 {
  font-size: 1.2em;
}

h6.emotion-0 {
  font-size: 1em;
}

<h2
  className=" emotion-0"
>
  Header text
</h2>
`;

exports[`UI Component: Header should correctly render h3 for level 3 1`] = `
.emotion-0 {
  margin: .5em 0;
  display: block;
  font-weight: bold;
  color: #0074ea;
}

h1.emotion-0,
h2.emotion-0 {
  color: #0074ea;
}

h1.emotion-0 {
  font-size: 2.4em;
}

h2.emotion-0 {
  font-size: 2em;
}

h3.emotion-0 {
  font-size: 1.8em;
}

h4.emotion-0 {
  font-size: 1.25em;
}

h5.emotion-0 {
  font-size: 1.2em;
}

h6.emotion-0 {
  font-size: 1em;
}

<h3
  className=" emotion-0"
>
  Header text
</h3>
`;

exports[`UI Component: Header should correctly render h4 for level 4 1`] = `
.emotion-0 {
  margin: .5em 0;
  display: block;
  font-weight: bold;
  color: #0074ea;
}

h1.emotion-0,
h2.emotion-0 {
  color: #0074ea;
}

h1.emotion-0 {
  font-size: 2.4em;
}

h2.emotion-0 {
  font-size: 2em;
}

h3.emotion-0 {
  font-size: 1.8em;
}

h4.emotion-0 {
  font-size: 1.25em;
}

h5.emotion-0 {
  font-size: 1.2em;
}

h6.emotion-0 {
  font-size: 1em;
}

<h4
  className=" emotion-0"
>
  Header text
</h4>
`;

exports[`UI Component: Header should correctly render h5 for level 5 1`] = `
.emotion-0 {
  margin: .5em 0;
  display: block;
  font-weight: bold;
  color: #0074ea;
}

h1.emotion-0,
h2.emotion-0 {
  color: #0074ea;
}

h1.emotion-0 {
  font-size: 2.4em;
}

h2.emotion-0 {
  font-size: 2em;
}

h3.emotion-0 {
  font-size: 1.8em;
}

h4.emotion-0 {
  font-size: 1.25em;
}

h5.emotion-0 {
  font-size: 1.2em;
}

h6.emotion-0 {
  font-size: 1em;
}

<h5
  className=" emotion-0"
>
  Header text
</h5>
`;

exports[`UI Component: Header should correctly render h6 for level 6 1`] = `
.emotion-0 {
  margin: .5em 0;
  display: block;
  font-weight: bold;
  color: #0074ea;
}

h1.emotion-0,
h2.emotion-0 {
  color: #0074ea;
}

h1.emotion-0 {
  font-size: 2.4em;
}

h2.emotion-0 {
  font-size: 2em;
}

h3.emotion-0 {
  font-size: 1.8em;
}

h4.emotion-0 {
  font-size: 1.25em;
}

h5.emotion-0 {
  font-size: 1.2em;
}

h6.emotion-0 {
  font-size: 1em;
}

<h6
  className=" emotion-0"
>
  Header text
</h6>
`;
